src/database/queries.ts

```diff
@@ -118,47 +202,147 @@ export const createVoucher = async (voucher: VoucherCreate): Promise<Uuid> => {
+    console.warn('(getTransaction)', e);
+    throw e;
+  }
+};
 
 /**
- * Setter function to update a Voucher's VendorUuid
+ * Helper for calculating a Transaction's value.
+ *
+ * Returns the sum in cents of all vouchers in the given array.
  */
-export const setVoucherVendorUuid = async (uuid: Uuid, vendorUuid: Uuid) =>
-  updateVoucher({ uuid, vendorUuid });
+export const calculateValue = async (voucherArray: number[]) => {
+  try {
+    let value = 0;
+    await Promise.all(
+      voucherArray.map(async v => {
+        const voucher = await getVoucher(v);
+        value += voucher.value;
+      }),
+    );
+    return value;
+  } catch (e) {
+    // eslint-disable-next-line no-console
+    console.warn('(calculateValue)', e);
+    throw e;
+  }
+};
 
 /**
- * Fetch all vouchers for a given vendor.
+ * Query to create a new transaction in Firebase.
+ *
+ * Parameters: a json with fields
+ *
+ *    `status`: the payment status of a transaction
+ *
+ *    `voucherArray`: an array of numbers representing voucher serial numbers
+ *
+ *    `vendorUuid`: current user's Uuid
+ *
+ * Returns the doc id if the query executes successfully.
  */
-export const getVouchersByVendorUuid = async (
+export const createTransaction = async (
+  transaction: TransactionCreate,
+): Promise<Uuid> => {
+  try {
+    const docRef = await addDoc(transactionCollection, transaction);
+    const value = await calculateValue(transaction.voucherArray);
+    await updateDoc(docRef, {
+      uuid: docRef.id,
+      timestamp: new Timestamp(Math.floor(Date.now() / 1000), 0),
+      value,
+    });
+    return docRef.id;
+  } catch (e) {
+    // eslint-disable-next-line no-console
+    console.warn('(createTransaction)', e);
+    throw e;
+  }
+};
+
+/**
+ * Helper for all Transaction setter functions.
+ */
+const updateTransaction = async (transaction: Partial<Transaction>) => {
+  try {
+    const docRef = doc(voucherCollection, transaction.uuid);
+    await updateDoc(docRef, transaction);
+  } catch (e) {
+    // eslint-disable-next-line no-console
+    console.warn('(updateTransaction)', e);
+    throw e;
+  }
+};
+
+/**
+ * Setter function to update a Transaction's status.
+ */
+export const setTransactionStatus = async (
+  uuid: Uuid,
+  status: TransactionStatus,
+) => updateTransaction({ uuid, status });
+
+/**
+ * Setter function to update a Transaction's voucherArray.
+ */
+export const setTransactionVoucherArray = async (
+  uuid: Uuid,
+  voucherArray: number[],
+) => {
+  const value = await calculateValue(voucherArray);
+  updateTransaction({ uuid, voucherArray, value });
+};
+
+/**
+ * Fetch all transactions for a given vendor.
+ *
+ * Returns an array of Transaction objects.
+ */
+export const getTransactionsByVendorUuid = async (
   vendorUuid: Uuid,
-): Promise<Voucher[]> => {
+): Promise<Transaction[]> => {
   try {
     const dbQuery = query(
-      voucherCollection,
+      transactionCollection,
       where('vendor_uuid', '==', vendorUuid),
```

Comment:
On the vendors list demo, it wasn't actually displaying the vouchers by vendor, and I traced it to this: it should be `vendorUuid` so it matches how we have it in Firebase

Reply:
Same with `getVouchersByVendorUuid`